src/Simplex/Chat/Store.hs

```diff
@@ -336,6 +337,24 @@ getUserContacts st User {userId} =
     contactNames <- liftIO $ map fromOnly <$> DB.query db "SELECT local_display_name FROM contacts WHERE user_id = ?" (Only userId)
     rights <$> mapM (runExceptT . getContact_ db userId) contactNames
 
+getNullContactConnections :: StoreMonad m => SQLiteStore -> User -> m [Connection]
+getNullContactConnections st User {userId} =
+  liftIOEither . withTransaction st $ \db ->
+    connections
+      <$> DB.queryNamed
+        db
+        [sql|
+          SELECT c.connection_id, c.agent_conn_id, c.conn_level, c.via_contact,
+            c.conn_status, c.conn_type, c.contact_id, c.group_member_id, c.created_at
+          FROM connections c
+          WHERE c.user_id = :user_id
+            AND c.conn_type = 'contact'
+            AND c.contact_id IS NULL
+        |]
+        [":user_id" := userId]
+  where
+    connections rows = Right $ map toConnection rows
```

Comment:
... and you won't need Right here, as it is never Left - so you can just put `map toConnection` above without creating this function.